src/index.js

```diff
@@ -188,9 +190,53 @@ class IOpipeWrapperClass {
       this.originalContext.done(err, data);
     });
   }
+  metric(keyInput, valueInput) {
+    const key = String(keyInput);
+    var numberValue = undefined;
+    var stringValue = undefined;
+    if (key.length >= 256) {
+      console.warn(
+        `Metric with key name ${key} is longer than allowed length of 256, metric will not be saved`
+      );
+      return;
+    }
+    if (valueInput === undefined) {
+      console.warn(
+        `Metric with key name ${key} has undefined value and will not be saved`
+      );
+      return;
+    }
+    if (typeof valueInput === 'number') {
```

Comment:
`Number.isFinite` may be the better check here, as `NaN` is typeof `number` (yayyy)

Reply:
Will update, and add a test for it